### PR TITLE
[shopsys] Create components for forms and restyle forms on eshop

### PR DESCRIPTION
**Pull request by VaniaToper**
| Q             | A
| ------------- | ---
|Description, reason for the PR| This PR implements new design for forms and form pages.
|New feature| No
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://ih-ssp-2305-restyle-forms.odin.shopsys.cloud
  - https://cz.ih-ssp-2305-restyle-forms.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Forms/Lib/FormLine.tsx [hunk 1]
```diff
@@ -6,10 +6,29 @@ type NativeProps = ExtractNativePropsFromDefault<HTMLAttributes<HTMLDivElement>,
...
+    ...props
+}) => (
+    <div
+        className={twMergeCustom(
+            'flex-1',
+            bottomGap && 'pb-3',
+            isHalfWidthInput && 'vl:w-1/2 vl:pr-1.5',
+            isSmallInput && 'w-[150px] flex-none vl:w-[220px]',
+            className,
+            'form-line',
```
Comment (sebaholesz) [inline project-base/storefront/components/Forms/Lib/FormLine.tsx]: Do you still need `form-line`?

File: project-base/storefront/components/Pages/Contact/ContactContent.tsx [hunk 2]
```diff
@@ -62,72 +61,60 @@ export const ContactContent: FC = () => {
...
-                            render={(checkbox) => <ChoiceFormLine>{checkbox}</ChoiceFormLine>}
-                            checkboxProps={{
-                                label: formMeta.fields.privacyPolicy.label,
-                                required: true,
-                            }}
-                        />
-                        <SubmitButton isWithDisabledLook={!formProviderMethods.formState.isValid}>
-                            {t('Send message')}
-                        </SubmitButton>
+                        <FormWrapper>
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Contact/ContactContent.tsx]: Why is FormWrapper not wrapping the form?

File: project-base/storefront/components/Pages/Registration/RegistrationContent.tsx [hunk 3]
```diff
@@ -47,61 +47,48 @@ export const RegistrationContent: FC = () => {
...
+                                }}
+                            />
+                            <FormButtonWrapper>
+                                <SubmitButton tid={TIDs.registration_submit_button}>{t('Sign up')}</SubmitButton>
+                            </FormButtonWrapper>
+                        </FormBlockWrapper>
+                    </FormWrapper>
                 </Form>
             </FormProvider>
-        </SimpleLayout>
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Registration/RegistrationContent.tsx:105]: Do we need `SimpleLayout` for anything?

File: project-base/storefront/components/Pages/OrderConfirmation/RegistrationAfterOrder.tsx [hunk 4]
```diff
@@ -115,44 +114,47 @@ export const RegistrationAfterOrder: FC = () => {
...
-                                    label: formMeta.fields.privacyPolicy.label,
-                                }}
-                                render={(checkbox) => (
-                                    <div className="mb-7">
-                                        <ChoiceFormLine>{checkbox}</ChoiceFormLine>
-                                    </div>
-                                )}
-                            />
+                    <FormProvider {...formProviderMethods}>
+                        <Form
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/OrderConfirmation/RegistrationAfterOrder.tsx:118]: Why did we swap the hierarchy here?

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationUser.tsx [hunk 5]
```diff
@@ -69,6 +46,19 @@ export const ContactInformationUser: FC = () => {
...
+                render={(textInput) => <FormLine>{textInput}</FormLine>}
+                textInputProps={{
+                    label: formMeta.fields.telephone.label,
+                    required: true,
+                    type: 'tel',
+                    autoComplete: 'tel',
+                    onChange: (event) => updateContactInformation({ telephone: event.currentTarget.value }),
+                }}
+            />
+        </FormBlockWrapper>
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationUser.tsx:62]: Why was the order of form elements changed?

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationEmail.tsx [hunk 6]
```diff
@@ -62,16 +60,19 @@ export const ContactInformationEmail: FC = () => {
...
+                        tid={TIDs.login_in_order_button}
+                        type="button"
+                        onClick={openLoginPopup}
+                    >
+                        {t('Sign in')}
+                    </Button>
+                </div>
             )}
-        </>
+        </FormBlockWrapper>
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationEmail.tsx:76]: Why was this changed? I thought we wanted the entire content as a button

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationSendOrderButton.tsx [hunk 7]
```diff
@@ -44,7 +44,17 @@ export const ContactInformationSendOrderButton: FC = () => {
...
+                <CheckboxControlled
+                    control={formProviderMethods.control}
+                    formName={formMeta.formName}
+                    name={formMeta.fields.isWithoutHeurekaAgreement.name}
+                    render={(checkbox) => <ChoiceFormLine>{checkbox}</ChoiceFormLine>}
+                    checkboxProps={{
+                        label: formMeta.fields.isWithoutHeurekaAgreement.label,
+                    }}
+                />
+            )}
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationSendOrderButton.tsx:57]: Why was it moved here?

**Review by sebaholesz — CHANGES_REQUESTED** [review]

File: project-base/storefront/components/Forms/Lib/FormLine.tsx (hunk 1 above)
Comment (VaniaToper) [reply project-base/storefront/components/Forms/Lib/FormLine.tsx]: No we don't, thanks

File: project-base/storefront/components/Pages/Contact/ContactContent.tsx (hunk 2 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/Contact/ContactContent.tsx]: Good point, first I'll explain why we don't put these styles to the Form component:
We don't always need to wrap the entire form with this component, for example ContactInformationWrapper

So it's just about naming, do you have a better name for this component in your mind?

File: project-base/storefront/components/Pages/Registration/RegistrationContent.tsx (hunk 3 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/Registration/RegistrationContent.tsx:105]: There still some pages with this layout, I will remove it when I will be styling stores page, because it will be the last which needs it

File: project-base/storefront/components/Pages/OrderConfirmation/RegistrationAfterOrder.tsx (hunk 4 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/OrderConfirmation/RegistrationAfterOrder.tsx:118]: There was a bad hierarchy compared to the other places on the project, so I unified it

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationUser.tsx (hunk 5 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationUser.tsx:62]: It was inspired by breno and agreed by Martin

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationEmail.tsx (hunk 6 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationEmail.tsx:76]: It was inspired by breno and agreed by Martin
But the page will be restyling in the future, so maybe Martin will suggest something else 🤔

File: project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationSendOrderButton.tsx (hunk 7 above)
Comment (VaniaToper) [reply project-base/storefront/components/Pages/Order/ContactInformation/FormBlocks/ContactInformationSendOrderButton.tsx:57]: Because we want to have all checkboxes in one place

File: project-base/storefront/components/Pages/Contact/ContactContent.tsx (hunk 2 above)
Comment (sebaholesz) [reply project-base/storefront/components/Pages/Contact/ContactContent.tsx]: I mean, you could name it `FormContentWrapper` to show that it wraps the content, but I do not mind the current name either. Up to you. You can resolve as you wish.

**Comment by sebaholesz** [comment]
You are missing upgrade notes

**Comment by sebaholesz** [comment]
Commit messages should be lowercase

**Review by sebaholesz — APPROVED** [review]
